artistools/packets/__init__.py

```diff
@@ -116,7 +151,8 @@ def emtrue_timestep(packet):
     return dfpackets
 
 
-def readfile_text(packetsfile):
+def readfile_text(packetsfile, modelpath=Path('.')):
+    columns = get_column_names(modelpath)
     try:
         inputcolumncount = len(pd.read_csv(packetsfile, nrows=1, delim_whitespace=True, header=None).columns)
```

Comment:
If you've read the column names from the artis source code, then the number of columns in the packets*.out should strictly match the number of inferred columns. Currently, the code allows the number of named columns to be longer than the actual columns in the file (for backwards compatibility).

Reply:
Do you want a check added for that?

Reply:
Yes. Easiest way would be to have get_column_names() return columns, from_sourcecode

If from_sourcecode:
    assert len(columns) == inputcolumncount